xonsh/ptk/completer.py

```diff
@@ -32,5 +36,25 @@ def get_completions(self, document, complete_event):
                                                      begidx,
                                                      endidx,
                                                      self.ctx)
+            if len(completions) <= 1:
+                pass
+            elif len(os.path.commonprefix(completions)) == len(prefix):
```

Comment:
So this is for the case where there is a common beginning to all possible matches that is longer than the typed in prefix.  Pressing tab in these cases will complete the common beginning without bringing up the menu.  Without this check, the space would be reserved without the menu popping up.  I think that if this is changed to a `<=` then this will still work for the cases you bring up.